modules/core/src/v2/coins/dot.ts

```diff
@@ -0,0 +1,249 @@
+   */
+  valuelessTransferAllowed(): boolean {
+    return true;
+  }
+
+  /**
+   * Generate ed25519 key pair
+   *
+   * @param seed
+   * @returns {Object} object with generated pub, prv
+   */
+  generateKeyPair(seed?: Buffer): KeyPair {
+    const keyPair = seed
+      ? accountLib.Dot.Utils.default.keyPairFromSeed(new Uint8Array(seed))
+      : new accountLib.Dot.KeyPair();
+    const keys = keyPair.getKeys();
+    if (!keys.prv) {
+      throw new Error('Missing prv in key generation.');
+    }
+    return {
+      pub: keys.pub,
+      prv: keys.prv + keys.pub,
+    };
+  }
+
+  /**
+   * Return boolean indicating whether input is valid public key for the coin.
+   *
+   * @param {String} pub the pub to be checked
+   * @returns {Boolean} is it valid?
+   */
+  isValidPub(pub: string): boolean {
+    return accountLib.Dot.Utils.default.isValidPublicKey(pub);
+  }
+
+  /**
+   * Return boolean indicating whether the supplied private key is a valid dot private key
+   *
+   * @param {String} prv the prv to be checked
+   * @returns {Boolean} is it valid?
+   */
+  isValidPrv(prv: string): boolean {
+    return accountLib.Dot.Utils.default.isValidPrivateKey(prv);
+  }
+
+  /**
+   * Return boolean indicating whether input is valid public key for the coin
+   *
+   * @param {String} address the pub to be checked
+   * @returns {Boolean} is it valid?
+   */
+  isValidAddress(address: string): boolean {
+    return accountLib.Dot.Utils.default.isValidAddress(address);
+  }
+
+  /**
+   * Explain/parse transaction
+   * @param params
+   * @param callback
+   */
+  explainTransaction(
+    params: ExplainTransactionOptions,
+    callback?: NodeCallback<any>
+  ): Bluebird<any> {
+    throw new MethodNotImplementedError('Dot recovery not implemented');
+  }
+
+  verifySignTransactionParams(params: SignTransactionOptions): VerifiedTransactionParameters {
+    const prv = params.prv;
+    const addressVersion = params.txPrebuild.addressVersion;
+
+    const txHex = params.txPrebuild.txHex;
+
+    if (_.isUndefined(txHex)) {
+      throw new Error('missing txPrebuild parameter');
+    }
+
+    if (!_.isString(txHex)) {
+      throw new Error(`txPrebuild must be an object, got type ${typeof txHex}`);
+    }
+
+    if (_.isUndefined(prv)) {
+      throw new Error('missing prv parameter to sign transaction');
+    }
+
+    if (!_.isString(prv)) {
+      throw new Error(`prv must be a string, got type ${typeof prv}`);
+    }
+
+    if (!_.has(params.txPrebuild, 'key')) {
+      throw new Error('missing public key parameter to sign transaction');
+    }
+
+    if (!_.isNumber(addressVersion)) {
+      throw new Error('missing addressVersion parameter to sign transaction');
+    }
+
+    // if we are receiving addresses do not try to convert them
+    const signer = !accountLib.Dot.Utils.default.isValidAddress(params.txPrebuild.key)
+      ? new accountLib.Dot.KeyPair({ pub: params.txPrebuild.key }).getAddress()
+      : params.txPrebuild.key;
```

Comment:
You can switch the conditional validation to be non-negative (remove the `!`) and easier to read by swapping the if/else content